flink-runtime/src/main/java/org/apache/flink/runtime/jobgraph/SavepointConfigOptions.java

```diff
@@ -46,4 +46,14 @@
                     .withDescription(
                             "Allow to skip savepoint state that cannot be restored. "
                                     + "Allow this if you removed an operator from your pipeline after the savepoint was triggered.");
+    /**
+     * Describes the mode how Flink should restore from the given savepoint or retained checkpoint.
+     */
+    public static final ConfigOption<RestoreMode> RESTORE_MODE =
+            key("execution.savepoint-restore-mode")
+                    .enumType(RestoreMode.class)
+                    .defaultValue(RestoreMode.NO_CLAIM)
+                    .withDescription(
+                            "Describes the mode how Flink should restore from the given"
+                                    + " savepoint or retained checkpoint.");
```

Comment:
Since we choose `NO_CLAIM` as default mode, I think we deserve to describe the difference compared with previous mode. Otherwise, users could face the problem of increasing file system space occupation after several rounds of launching jobs.

Reply:
I intentionally did not want to describe it yet, because the `NO_CLAIM` mode is not implemented in this PR, as described in the FLIP.

However, it made me think that it's probably a better approach to have the `LEGACY` mode explicitly at least at this point in time. Therefore I renamed `NO_CLAIM` to `LEGACY` and extended the description a bit.

Reply:
I can see that the LEGACY mode is totally the same as current restoring mode in Flink, and the name also indicates that the behavior is the same as before. However, how can we tell users that the new 1st checkpoint would be a full checkpoint after restoring? With the name of LEGACY, users would still think the new 1st checkpoint is an incremental one.

Reply:
Yes, you are right `LEGACY` is the current behaviour. The other mode `NO_CLAIM` is not part of this PR. It will come in the next PR. 

> With the name of LEGACY, users would still think the new 1st checkpoint is an incremental one.
And they would be right. The `LEGACY` mode is the current behaviour, with no changes at all.

Reply:
side note: Personally, I am not 100% convinced we should have the `LEGACY` mode in the final version (once we add proper `NO_CLAIM` mode). We might remove it in the end, however I want to add it for now to have the master branch releasable at all times.

Reply:
Got it. This sounds more reasonable.